File: spec/java/src/io/kaitai/struct/spec/TestDebugSwitchUser.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.DebugSwitchUser;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestDebugSwitchUser extends CommonSpec {

    @Test
    public void testDebugSwitchUser() throws Exception {
        DebugSwitchUser r = DebugSwitchUser.fromFile(SRC_DIR + "nav_parent_switch.bin");

        // --debug implies --no-auto-read
        r._read();

        assertIntEquals(r.code(), 1);
        assertIntEquals(((DebugSwitchUser.One) (r.data())).val(), -190);
    }
}

File: spec/python/spec/test_debug_switch_user.py
import unittest
from testformats.debug_switch_user import DebugSwitchUser

class TestDebugSwitchUser(unittest.TestCase):
    def test_debug_switch_user(self):
        with DebugSwitchUser.from_file('src/nav_parent_switch.bin') as r:
            # --debug implies --no-auto-read
            r._read()

            self.assertEqual(r.code, 1)
            self.assertEqual(r.data.val, -190)
